frontends/election-manager/src/components/export_logs_modal.tsx

```diff
@@ -0,0 +1,283 @@
+import React, { useContext, useEffect, useState } from 'react';
+import path from 'path';
+import { usbstick, throwIllegalValue } from '@votingworks/utils';
+
+import assert from 'assert';
+import { LogEventId } from '@votingworks/logging';
+import { AppContext } from '../contexts/app_context';
+import { Modal } from './modal';
+import { Button } from './button';
+import { Prose } from './prose';
+import { LinkButton } from './link_button';
+import { Loading } from './loading';
+import { MainChild } from './main';
+import { UsbImage } from './save_file_to_usb';
+
+const { UsbDriveStatus } = usbstick;
+
+const LOGS_ROOT_LOCATION = '/var/log';
+const LOG_NAME = 'vx-logs.log';
+
+export interface Props {
+  onClose: () => void;
+}
+
+enum ModalState {
+  ERROR = 'error',
+  SAVING = 'saving',
+  DONE = 'done',
+  INIT = 'init',
+}
+
+export function ExportLogsModal({ onClose }: Props): JSX.Element {
+  const { usbDriveStatus, currentUserSession, logger } = useContext(AppContext);
+  assert(currentUserSession); // TODO(auth) should this check for a specific user type
```

Comment:
yes, in the future we'll want an admin session of some kind here, and we may want to reauth in case the card has been removed.